mlir/lib/Target/LLVMIR/ModuleTranslation.cpp

```diff
@@ -632,8 +632,25 @@ llvm::Constant *mlir::LLVM::detail::getLLVMConstant(
           llvm::ElementCount::get(numElements, /*Scalable=*/isScalable), child);
     if (llvmType->isArrayTy()) {
       auto *arrayType = llvm::ArrayType::get(elementType, numElements);
-      SmallVector<llvm::Constant *, 8> constants(numElements, child);
-      return llvm::ConstantArray::get(arrayType, constants);
+      if (child->isZeroValue()) {
+        return llvm::ConstantAggregateZero::get(arrayType);
+      } else {
+        if (llvm::ConstantDataSequential::isElementTypeCompatible(elementType)) {
+          // TODO: Handle all compatible types. This code only handle i32.
```

Comment:
```suggestion
          // TODO: Handle all compatible types. This code only handles i32.
```

Reply:
Thanks for the review. I'm not sure we want to keep the `ConstantDataArray` path. I'm happy to add more types here if we want to keep it in this PR.

Reply:
I suspect we want to have it for efficiency reasons. We have had compile time issues with large nested constants previously that I suspect using ConstantData avoids. This is perfectly fine for a separate PR.

Reply:
Ok! I have added at least all int types supported.